File: backend/routes/admission.routes.js
import { Router } from 'express';
import {
  getAdmissions,
  getAdmissionById,
  createAdmission,
  updateAdmissionStatus,
  deleteAdmission,
} from '../controllers/admission.controller.js';
import { verifyToken, requireRole } from '../middleware/auth.middleware.js';

const router = Router();

// Public route – anyone can submit an application
router.post('/', createAdmission);

// Protected routes – require authentication
router.get('/',    verifyToken, requireRole('admin'), getAdmissions);
router.get('/:id', verifyToken, requireRole('admin'), getAdmissionById);
router.put('/:id/status', verifyToken, requireRole('admin'), updateAdmissionStatus);
router.delete('/:id', verifyToken, requireRole('admin'), deleteAdmission);

export default router;

File: backend/routes/admission_routes.py
from flask import Blueprint
from controllers.admission_controller import (
    get_admissions,
    get_admission_by_id,
    create_admission,
    update_admission_status,
    delete_admission,
)
from middleware.auth_middleware import verify_token, require_role

admission_bp = Blueprint('admissions', __name__)

# Public route – anyone can submit an application
admission_bp.route('/', methods=['POST'])(create_admission)

# Protected routes – require authentication
admission_bp.route('/', methods=['GET'])(
    verify_token(require_role('admin')(get_admissions))
)
admission_bp.route('/<string:admission_id>', methods=['GET'])(
    verify_token(require_role('admin')(get_admission_by_id))
)
admission_bp.route('/<string:admission_id>/status', methods=['PUT'])(
    verify_token(require_role('admin')(update_admission_status))
)
admission_bp.route('/<string:admission_id>', methods=['DELETE'])(
    verify_token(require_role('admin')(delete_admission))
)
